yjit/src/codegen.rs

```diff
@@ -335,65 +335,71 @@ fn record_global_inval_patch(cb: &mut CodeBlock, outline_block_target_pos: CodeP
     CodegenGlobals::push_global_inval_patch(cb.get_write_ptr(), outline_block_target_pos);
 }
 
-/*
 // Verify the ctx's types and mappings against the compile-time stack, self,
 // and locals.
-static void
-verify_ctx(jitstate_t *jit, ctx_t *ctx)
+fn verify_ctx(jit: &JITState, ctx:&Context)
 {
-    // Only able to check types when at current insn
-    RUBY_ASSERT(jit_at_current_insn(jit));
+    if get_option!(verify_ctx) {
+        // Only able to check types when at current insn
+        assert!(jit_at_current_insn(jit));
```

Comment:
I think you should put this check in the caller to make it more visible that this only happens with this command-line option.

Reply:
Okay. If we ever call verify_context() again we'll need to duplicate that check in each place, though.